src/software/utils/main_imageProcessing.cpp

```diff
@@ -469,6 +469,82 @@ void processImage(image::Image<image::RGBAfColor>& image, const ProcessingParams
             "Unsupported mode! If you intended to use a non-local means filter, please add OpenCV support.");
 #endif
     }
+
+    if (pParams.applyDcpMetadata)
+    {
+        bool dcpMetadataOK = !imageMetadata["AliceVision:DCP:Temp1"].empty() && !imageMetadata["AliceVision:DCP:Temp2"].empty() &&
+                             !imageMetadata["AliceVision:DCP:ForwardMatrixNumber"].empty() && !imageMetadata["AliceVision:DCP:ColorMatrixNumber"].empty();
+
+        int colorMatrixNb;
+        int fwdMatrixNb;
+
+        if (dcpMetadataOK)
+        {
+            colorMatrixNb = atoi(imageMetadata["AliceVision:DCP:ColorMatrixNumber"].c_str());
```

Comment:
Use `std::atoi`.
The C implementation does not throw an error in case of failure in contrast to the C++ version.

https://cplusplus.com/reference/string/stoi/

https://cplusplus.com/reference/cstdlib/atoi/ => "No-throw guarantee: this function never throws exceptions. If str does not point to a valid C-string, or if the converted value would be out of the range of values representable by an int, it causes undefined behavior."

Reply:
Done